interp/type.go

```diff
@@ -198,11 +198,16 @@ func nodeType(interp *Interpreter, sc *scope, n *node) (*itype, error) {
 						t.incomplete = true
 					}
 				} else {
+					c0 := n.child[0]
 					// Evaluate constant array size expression
-					if _, err = interp.cfg(n.child[0], sc.pkgID); err != nil {
+					if _, err = interp.cfg(c0, sc.pkgID); err != nil {
 						return nil, err
 					}
-					t.incomplete = true
+					if v, ok := c0.rval.Interface().(constant.Value); ok {
+						t.size = constToInt(v)
+					} else {
+						t.incomplete = true
+					}
```

Comment:
We are in a switch, so we can break out early on error, like so:

```suggestion
					if v, ok := c0.rval.Interface().(constant.Value); !ok {
						t.incomplete = true
						break
					}
					t.size = constToInt(v)
```